### ISSUE TITLE
Include field for human-readable check info

### ISSUE THREAD

**Issue by bgro** [scapcommunity/XCCDF#14]
CIS benchmarks viewed in PDF, Word or XLSX contain a human-readable description of how to audit a setting, which is not contained in the XCCDF version of the guide, simply because there is no field for this type of content. 

XCCDF should therefore carry a `checktext` element akin to the the `fixtext` element.

**Comment by redhatrises** [comment]
One concern that I have with is, is that this duplicates what OCIL is and does which already exists in the XCCDF and datastreams.

**Comment by wmunyan** [comment]
I'm not sure @bgro 's idea duplicates anything provided by OCIL.  CIS adds basically the "manual" way to check the configuration setting as informational text for someone reading the prose.  We shoe-horn this info into either `description` or `rationale` elements to make the prose read correctly.  It's not asking the user to _do_ any of those things to pass the rule, though, so its not really interactive with the user.

I agree that a `checktext` element could be a useful addition.

**Comment by redhatrises** [comment]
Ahhh.... but it is duplicate in the sense that it can (and often does) provide the same information. Also, why are we encouraging manual checks as part of the XCCDF when in reality, again, this is what OCIL is really for?

**Comment by redhatrises** [comment]
Basically arguing for "Is this the time to start pushing OCIL" since it is underutilized and theoretically used for this.

**Comment by bgro** [comment]
Nothing against pushing OCIL, but it seems to me that for satisifying a very simple use-case (namely providing human-readable inforamtion of how a check can be performed), it is overkill to require the author to resort to OCIL (and the  consumer to parse OCIL). 

So in Compliance-as-Code, you have in your rule file things like

~~~
ocil_clause: 'the above command returns no output, or FAIL_DELAY is configured less than the expected value'

ocil: |-
    Verify the <tt>FAIL_DELAY</tt> setting is configured correctly in the <tt>/etc/login.defs</tt> file by
    running the following command:
    <pre>$ sudo grep -i "FAIL_DELAY" /etc/login.defs</pre>
    All output must show the value of <tt>FAIL_DELAY</tt> set as shown in the below:
    <pre>$ sudo grep -i "FAIL_DELAY" /etc/login.defs
    fail_delay <sub idref="var_accounts_fail_delay" /></pre>
~~~

If you are more happy to call the field `ocil` rather than `checktext`, then I am also fine with that :)

But I think XCCDF should provide an easy way to carry that basic information. If SCAP generating mechanisms than  use that field for automated generation of OCIL (or maybe add a possibility in OCIL to refer to the `checktext` (or whatever it is called) field in the XCCDF), even better!

**Comment by wmunyan** [comment]
The intent of the text in the CIS benchmarks is to illustrate to the reader how they might manually check a particular setting.  It isnt meant to be interactive at all, nor to suggest that the user needs to perform the manual check in order to pass the Rule.  It is simply additional information provided to the user, similar to what the `fixtext` field provides.  I dont think the intent of this issue is to push OCIL or replace OCIL or do anything with OCIL.  It's merely defining another textual element to the prose.

**Comment by DavidRies** [comment]
I agree that this would be useful and does not overlap/conflict significantly with the OCIL use cases. Many content authors already include this. If there was an explicit, optional place for it, then we could better leverage it in reports, UIs, etc. 

In my experience this is not used as an alternative to the automated checks. It's used by content developers debugging content issues and end-users investigating results from automated checks.

**Comment by redhatrises** [comment]
@DavidRies thanks that description response is useful and very helpful in understanding the request.

**Comment by bgro** [comment]
I think this was not looked at during the workshop right? Do we have agreement that there should be an `checktext` akin to `fixtext`?

**Comment by bgro** [comment]
Please refer to 

https://github.com/scapcommunity/XCCDF/pull/31/files

for a suggestion of how to add to the XCCDF spec.

**Comment by bgro** [comment]
To my understanding, this issue is done and ready for "NISTification".
